examples/lin_algebra/gmres.cpp

```diff
@@ -0,0 +1,274 @@
+
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+#include <arrayfire.h>
+#include <af/util.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <vector>
+#include <iostream>
+using namespace af;
+
+struct info_solver {
+  std::vector<float> iter;
+  std::vector<float> time;
+  std::vector<float> nnz;
+};
+template <class T>
+void Update(array& x, const int k, const std::vector<T>&H,
+            const int R, const std::vector<T>&s,
+            const array v)
+{
+  T y[k+1];
```

Comment:
Use of variable length arrays is not technically valid C++ though GCC at least supports it anyway. I recommend using a vector instead.